homeassistant/components/weather/__init__.py

```diff
@@ -207,93 +531,200 @@ def precision(self) -> float:
     def state_attributes(self):
         """Return the state attributes, converted from native units to user-configured units."""
         data = {}
-        if self.temperature is not None:
-            data[ATTR_WEATHER_TEMPERATURE] = show_temp(
-                self.hass,
-                self.temperature,
-                self.temperature_unit,
-                self.precision,
-            )
+
+        precision = self.precision
+
+        if (temperature := (self.native_temperature or self.temperature)) is not None:
```

Comment:
This did not work as intended if `self.native_temperature` returned `0`, with the suggested changes above, we can simplify to:
```suggestion
        if (temperature := self.native_temperature) is not None:
```